File: medium/3871_count_commas_in_range_ii/solution.py
class Solution:
    def countCommas(self, n: int) -> int:
        commasCount = 0

        bound = 1000
        multiplier = 1

        while bound <= n:
            commasCount += (min(bound * 1000 - 1, n) - bound + 1) * multiplier
            bound *= 1000
            multiplier += 1

        return commasCount

File: medium/3871_count_commas_in_range_ii/solution.cpp
#include <algorithm>
class Solution {
public:
  long long countCommas(long long n) {
    long long commasCount = 0;

    for (long long bound = 1000, multiplier = 1; bound <= n;
         bound *= 1000, ++multiplier)
      commasCount += (std::min(bound * 1000 - 1, n) - bound + 1) * multiplier;

    return commasCount;
  }
};
